src/rp2_common/hardware_adc/include/hardware/adc.h

```diff
@@ -88,6 +88,16 @@ static inline void adc_select_input(uint input) {
     hw_write_masked(&adc_hw->cs, input << ADC_CS_AINSEL_LSB, ADC_CS_AINSEL_BITS);
 }
 
+/*! \brief  ADC get input
+ *  \ingroup hardware_adc
+ *
+ * Returns the currently selected input. 0...3 are GPIOs 26...29 respectively.
+ * Input 4 is the onboard temperature sensor.
+ */
+static inline uint adc_selected_input(void) {
```

Comment:
@kilograham obviously has the final decision, but I wonder if a function name of `adc_get_selected_input` would be slightly clearer? (In hindsight I guess it's a shame that `adc_select_input` wasn't named `adc_select_input_channel`? Because that would allow this function to be named `adc_get_input_channel`.)

Reply:
yes, thank adc_get_selected_input is the better name... i did go thru and tried to make all names logical and consistent but I may have missed a couple.